### PR TITLE
Migrate to the latest integration test suite

### LINKED ISSUE #30: The cross-language test suite should be up-to-date
The template was recently updated with minor changes. As a synchronization exercise all the Dredd hooks implementations are currently being updated and this one should too.

See for reference https://github.com/apiaryio/dredd-hooks-template/issues/16

### PR DESCRIPTION
**Pull request by honzajavorek**
Close https://github.com/apiaryio/dredd-hooks-ruby/pull/31, close https://github.com/apiaryio/dredd-hooks-ruby/issues/30

Related: https://github.com/apiaryio/dredd/issues/917#issuecomment-482189943

### CODE REVIEW

**Comment by honzajavorek** [comment]
I'm not skilled enough with Ruby to figure out how to make the template work. I think https://github.com/apiaryio/dredd-hooks-template/issues/4#issuecomment-495677500 is a blocker and needs to be resolved first.

**Comment by honzajavorek** [comment]
This should pass now once https://github.com/apiaryio/dredd-hooks-template/pull/65 is released. Thanks @abtris for all the help 🙏

File: .gitignore [hunk 1]
```diff
@@ -6,10 +6,13 @@
...
+/vendor/
 *.bundle
 *.so
 *.o
 *.a
 mkmf.log
 npm-debug.log
 *.gem
+node_modules
+package-lock.json
```
Comment (gonzalo-bulnes) [inline .gitignore:18]: I think the `Gemfile.lock` could be ignored in the same way the `package-lock.json` is ignored.

_I used to include the lock files in my packages for testing, but to be honest I haven't ever run into a situation where that proved useful to troubleshoot differences between local testing and continuous testing, so maybe it's not that useful after all. YMMV_

```suggestion
package-lock.json
Gemfile.lock
```

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
   - "2.2.4"
   - "2.4"
   # - "2.5" https://github.com/apiaryio/dredd-hooks-ruby/issues/36
   - "2.6"
+before_install:
+  - "nvm install node && nvm use node"
+  - "npm install"
+script:
+  - "export BUNDLE_GEMFILE=$(pwd)/Gemfile"
+  - "bundle exec rake && npm test"
```
Comment (gonzalo-bulnes) [inline .travis.yml:14]: All tests run successfully in Travis CI :heavy_check_mark:

File: Gemfile.lock [hunk 3]
```diff
@@ -6,36 +6,7 @@ PATH
 GEM
```
Comment (gonzalo-bulnes) [inline Gemfile.lock:6]: If it was to be added to the `.gitignore`, this file could be removed. (See comment on the `.gitignore` file.)

File: README.md [hunk 4]
```diff
@@ -89,20 +89,35 @@ Development
...
+A few [Cucumber][cucumber] features from the [dredd-hooks-template] provide an end-to-end test harness. The test suite is provided as a [Node.js] package and needs to be installed by `npm`:
+
+```
+npm install
+```
+
+Then the tests can be ran using the `npm test` command, but for the Ruby paths to be correctly set, an environment variable `BUNDLE_GEMFILE` needs to be set to the path to the `Gemfile`. An example how running the end-to-end tests can look like on macOS:
+
+```
+BUNDLE_GEMFILE=$(pwd)/Gemfile npm test
```
Comment (gonzalo-bulnes) [inline README.md:111]: I was able to run the test suite locally following these installation and execution instructions. :heavy_check_mark:

File: Rakefile [hunk 5]
```diff
@@ -29,5 +26,5 @@ rescue LoadError
   end
 end
 
-task default: ['spec:public', 'spec:development', :cucumber]
+task default: ['spec:public', 'spec:development']
```
Comment (gonzalo-bulnes) [inline Rakefile:29]: Cucumber is no longer part of the Ruby realm. :heavy_check_mark:

File: package.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "scripts": {
+    "test": "dredd-hooks-template test"
+  },
+  "private": true,
+  "devDependencies": {
+    "dredd-hooks-template": "1.1.3"
```
Comment (gonzalo-bulnes) [inline package.json:7]: Yes for the `test` command and `devDependency`. :heavy_check_mark: :heavy_check_mark:

File: lib/dredd_hooks/cli.rb [hunk 7]
```diff
@@ -10,6 +10,7 @@ def self.start(error=STDERR, out=STDOUT, files)
 
       # Run the server
       out.puts 'Starting Ruby Dredd Hooks Worker...'
+      out.flush
```
Comment (gonzalo-bulnes) [inline lib/dredd_hooks/cli.rb:13]: I'm guessing this relates to [waiting for the server to output "Starting"](https://github.com/apiaryio/dredd-hooks-ruby/pull/37/files#diff-60124dd45a23a8b2ea2358d2c420d09aR5). Ensuring the output seems reasonable to me. :white_check_mark:

**Review by gonzalo-bulnes — APPROVED** [review]
Except _optionally_ removing and ignoring the `Gemfile.lock`, I can't see anything missing or out of place. I'd say ship it @honzajavorek :slightly_smiling_face: 

I'm glad to see this project keeps moving, sorry for not being very responsive these days!

File: lib/dredd_hooks/cli.rb (hunk 7 above)
Comment (honzajavorek) [reply lib/dredd_hooks/cli.rb:13]: Exactly 👍

File: .gitignore (hunk 1 above)
Comment (honzajavorek) [reply .gitignore:18]: In the realm of Python they suggest to commit Pipfile.lock as well, but as you wrote - I usually don't see it that useful for libraries. I might do what you suggest as a next step in a separate PR.

**Comment by honzajavorek** [comment]
@gonzalo-bulnes Thank you very much for the review! It is nice to see you around again. It took me a while to carve out some time to clean the mess around the hooks template, so don't worry about not being responsive. I'll do my best what my limited Ruby abilities allow me to keep updating the Ruby hooks and if you feel like helping out, even with a review like you just did, it is all very appreciated 🙇

**Review by kylef — APPROVED** [review]

**Pull request merged**